sig: add description for expression SIG (#51)
pingcap/community: special-interest-groups/sig-expr/roles-and-organization-management.md

# SIG Expr Roles and Organization Management

> ## Join && Promotion

Reviewer: Why $$?

## Join

> The minimal requirement for contributors to join the Expr SIG is to have at

Reviewer: ```suggestion
The minimal requirement for contributors to join the Expr SIG is to have
```

> least the **Active Contributor** community role in the expression package.

Reviewer: > in the expression package

Can we say "in the expression package subproject"?

> least the **Active Contributor** community role in the expression package.

Reviewer: ```suggestion
the **Active Contributor** community role in the expression package.
```

> 

Reviewer: ```suggestion
## Promotion

```

### From Contributor to Active Contributor

> Requirement:

Reviewer: ```suggestion
Requirements:
```

> * Have at least 8 PRs get merged in the expression package within one year.

Reviewer: ```suggestion
* Have at least 8 PRs merged in the expression package subproject within one year.
```

> * Nominated by at least 2 Expr SIG Reviewers, Committer or Maintainer.

Reviewer: > 2 Expr SIG Reviewers, Committer or Maintainer.
 
1 Committer or 1 Maintainer?

### From Active Contributor to Reviewer

> Requirement:

Reviewer: ```suggestion
Requirements:
```

> * Have at least 30 PRs get merged in the expression package within one year.

Reviewer: ```suggestion
* Have at least 30 PRs merged in the expression package subproject within one year.
```

> * Nominated by at least 2 Committer or Maintainer.

Reviewer: ```suggestion
* Nominated by at least 2 Committers or 2 Maintainers.
```

### From Reviewer to Committer

Requirement:

* Have at least 60 PRs get merged in the expression package within one year.
* Have reviewed at least 20 PRs in the expression package within one year.

> * Nominated by at least 2 Committer or Maintainer.

Reviewer: The same edits as above

## Quit && Demotion

### Active Contributor

**Automatically retired**:

* No contribution to the expression package within one year.

**How to rejoin**:

> * Should meet the requirement of Active Contributor again.

Reviewer: ```suggestion
* Should meet the requirements for Active Contributor again.
```

### Reviewer

> Demoted to Active Contributor, satisifies at least 1 of the following

Reviewer: ```suggestion
For a Reviewer to be demoted to an Active Contributor,  he/she must satisfy at least 1 of the following conditions:
```

> scenarios:

Reviewer: ```suggestion:
```

> * Haven't reviewed or merged any PR in the expression package within 3 monthes.

Reviewer: ```suggestion
* Haven't reviewed or merged any PR in the expression package subproject within 3 months.
```

> * Not appeared in the SIG meeting without any reason continuously 3 times.

Reviewer: ```suggestion
* Haven't appeared in the SIG meeting without any reason for 3 times continuously.
```

> * At least 2 Committers or Maintainers thinks that Reviewer is not qualified.

Reviewer: ```suggestion
* At least 2 Committers or Maintainers agree that the Reviewer is not qualified.
```

> How to retain:

Reviewer: ```suggestion
How to retain the Reviewer role:
```

> * Reviewd or merged 5 PRs within 1 month

Reviewer: ```suggestion
* Have reviewed or merged 5 PRs within 1 month
```

> ## Voluntary Quit or Demotion

Reviewer: ```suggestion
## Voluntarily Quit or Demote
```

Please email to [Yanqing Zhang](mailto:zhangyanqing@pingcap.com) if you are

> going to voluntaryly quit or demotion

Reviewer: ```suggestion
going to quit or demote voluntarily.
```

# SIG Expr Roles and Organization Management

## Join and Promotion

## Join

The minimal requirement for contributors to join the Expr SIG is to have
the **Active Contributor** community role in the expression package.

## Promotion

### From `Contributor` to `Active Contributor`

Requirements:

* Have at least 8 PRs get merged in the expression package within one year.
* Nominated by at least 2 Expr SIG Reviewers, Committer or Maintainer.

### From `Active Contributor` to `Reviewer`

Requirements:

* Have at least 30 PRs get merged in the expression package within one year.
* Nominated by at least 2 Committers or 2 Maintainers.

### From `Reviewer` to `Committer`

Requirement:

* Have at least 60 PRs get merged in the expression package within one year.
* Have reviewed at least 20 PRs in the expression package within one year.

> * Nominated by at least 2 Committer or Maintainer.

Reviewer: ```suggestion
* Nominated by at least 2 Committers or Maintainers.
```

# SIG Expr Roles and Organization Management

## Roles

1. Development Role:

* Active Contributor
* Reviewer
* Committer
* Maintainer

2. Management Role:

* SIG Member
* SIG Technical Lead

## Join && Promotion

## Join

The minimal requirement for contributors to join the Expr SIG is to have at
least the **Active Contributor** community role in the expression package.

### From Contributor to Active Contributor

Requirement:

> * Have at least 8 PRs get merged in the expression package within one yeay.

Reviewer 2: yeay/year

Author: done
